Creation of an endpoint to register rooms

Benjamin-Chain opened the pull request:
## Summary
- Creation of service layer and the base structure
- Creation of controller layer and the base structure
- Creation of the method required to register a room
- Modified unit testing without Singleton

Benjamin-Chain committed:
Creating an endpoint to add rooms

gnievestm reviewed (changes requested):
Fix conflicts please

Benjamin-Chain committed:
Merge branch 'develop' into feature/TM23-116-EndpointToRegisterRoom

Benjamin-Chain commented:
@gnievestm solved

gnievestm committed:
Fix annotation and attributes

- The id of the room is not enough to determine 2 rooms are the same, given that name and address should be unique, then the id is not needed.
- Adapt code to work without the id.
- Fix unit tests
- Take out annotation in entity clases.

gnievestm committed:
Merge branch 'develop' into feature/TM23-116-EndpointToRegisterRoom
